### PR TITLE
Discretionary methods and definitions

### PR DESCRIPTION
**Pull request by vafl-brut**

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -193,3 +193,67 @@ export const enum ECARD_STATUS {
...
+
+export interface IDiscretionaryCampaignParticipant {
+    id: number,
+    first_name: string,
+    last_name: string
+}
+
+export interface IDiscretionaryLedgerTransaction extends IPodiumModel {
+    id: number,
+    // external_id: string,
```
Comment (snapjay) [inline types/index.d.ts]: why are theses commented

File: src/index.ts [hunk 2]
```diff
@@ -40,5 +43,7 @@ export class Podium {
         }
         this.Terms = new Terms(settings)
         this.Users = new Resource(settings).SetResource('user').SetLegacy(true)
+        this.Discretionary = new Discretionary(settings)
```
Comment (snapjay) [inline src/index.ts]: Make these grouped like ecard
Discretionary.Reports

File: src/Api/Discretionary.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+export class Discretionary extends Resource {
+
+    constructor(settings: ISettings) {
+        super(settings)
+    }
+
+    // Retrieves single campaign.
+    public GetCampaign(campaignId: number): IPodiumPromise<IDiscretionaryCampaign> {
+        super.SetResource(`campaign/discretionary/${campaignId}`)
```
Comment (snapjay) [inline src/Api/Discretionary.ts]: This is not necessary; inherited from Resource

Comment (vafl-brut) [reply src/Api/Discretionary.ts]: Resource overwrite makes it necessary (i think)

**Comment by vafl-brut** [comment]
PR updated

**Pull request merged**